Replace __GLIBC__ with __linux__ since musl also supports it
so checking __linux__ is more accomodating

See http://git.alpinelinux.org/cgit/aports/tree/community/webkit2gtk/musl-fixes.patch?id=219435d86d7e8fac9474344a7431c62bd2525184

Upstream-Status: Pending
Signed-off-by: Khem Raj <raj.khem@gmail.com>

Index: webkitgtk-2.12.1/Source/JavaScriptCore/heap/MachineStackMarker.cpp
===================================================================
--- webkitgtk-2.12.1.orig/Source/JavaScriptCore/heap/MachineStackMarker.cpp
+++ webkitgtk-2.12.1/Source/JavaScriptCore/heap/MachineStackMarker.cpp
@@ -566,7 +566,7 @@ void* MachineThreads::Thread::Registers:
 #error Unknown Architecture
 #endif
 
-#elif defined(__GLIBC__) && ENABLE(JIT)
+#elif defined(__linux__) && ENABLE(JIT)
 
 #if CPU(X86)
     return reinterpret_cast<void*>((uintptr_t) regs.machineContext.gregs[REG_ESP]);
@@ -665,7 +665,7 @@ void* MachineThreads::Thread::Registers:
 #error Unknown Architecture
 #endif
 
-#elif defined(__GLIBC__)
+#elif defined(__linux__) // glibc and musl
 
 // The following sequence depends on glibc's sys/ucontext.h.
 #if CPU(X86)
@@ -747,7 +747,7 @@ void* MachineThreads::Thread::Registers:
 #error Unknown Architecture
 #endif
 
-#elif defined(__GLIBC__)
+#elif defined(__linux__) // glibc and musl
 
 // The following sequence depends on glibc's sys/ucontext.h.
 #if CPU(X86)
@@ -838,7 +838,7 @@ void* MachineThreads::Thread::Registers:
 #error Unknown Architecture
 #endif
 
-#elif defined(__GLIBC__)
+#elif defined(__linux__) // glibc and musl
 
 // The following sequence depends on glibc's sys/ucontext.h.
 #if CPU(X86)
